vxfreeswitch/tests/test_voice.py

```diff
@@ -549,6 +550,30 @@ def create_worker(self, config={}):
         return self.tx_helper.get_transport(default)
 
     @inlineCallbacks
+    def wait_for_client_registration(self, worker, clientid):
+        '''Waits until the client has registered itself to the worker. Returns
+        False if it times out waiting.'''
+        for _ in range(5):
+            if worker._originated_calls.get(clientid, None) is None:
+                returnValue(True)
+            d = defer.Deferred()
+            reactor.callLater(0.01, d.callback, None)
+            yield d
```

Comment:
Resolved in https://github.com/praekelt/vumi-freeswitch-esl/pull/32/commits/781287523f75d9bd923a2ea2c8a10ac994c90163